### ISSUE TITLE
missing client token

### ISSUE THREAD

**Issue by messiahUA** [hashicorp/vault-k8s#75]
Using the latest vault injector 0.2.0 I've created a simple test pod and provided the corresponding serviceAccountName, but vault-agent-init container for some reason gives an error:
```
auth.handler: error authenticating: error="Error making API request.

URL: PUT https://vault.vault.svc:8200/v1/clusters/services/login
Code: 400. Errors:

* missing client token" backoff=1.6070817979999998
```
I've checked that service account token is mounted and is available at /var/run/secrets/kubernetes.io/serviceaccount/token

What may be the reason? I don't see any configuration parameters related to token or SA.

**Comment by tsurankov** [comment]
It seems like the correct path should be https://vault.vault.svc:8200/v1/auth/clusters/services/login
I have solved a similar issue with an external Vault server by overriding the env variable in the deploy/injector-deployment.yaml file:
`            - name: AGENT_INJECT_VAULT_AUTH_PATH
              value: "auth/clustername"`

**Comment by messiahUA** [comment]
@tsurankov indeed, thanks! I guess this issue can be closed unless there is an opportunity to simply add "auth/" by default to the path.

**Comment by shuklashashank** [comment]
I am also facing the same issue and I found the correct entry in injector yaml.
AGENT_INJECT_VAULT_AUTH_PATH: auth/kubernetes

**Comment by jimsnab** [comment]
And for me, using sidecar and GCP:

```
spec:
  template:
    metadata:
      annotations:
        vault.hashicorp.com/auth-path: "auth/gcp"
```
